lib/Server.js

```diff
@@ -39,18 +39,21 @@ if (!process.env.WEBPACK_SERVE) {
 }
 
 class Server {
-  constructor(compiler, options = {}) {
```

Comment:
We need keep compatibility here, just ignore compiler from constructor for apply

Reply:
Ideally after stable release v4, we will release 4.1/4.2 with plugin support, and for v5 we drop compiler from `constructor`

Reply:
Sounds good, we need do it before stable release, let's do it separately and fix in webpack-cli too, maybe adding hidden static option `_newConstructor`

Reply:
On it 👍🏽